### DISCUSSION TITLE
ID Reassignment in yolo tracker long term video

### DISCUSSION THREAD

**Discussion by barisatayy** [ultralytics/ultralytics#23731] [Q&A]
Hi,
I trained a YOLOv26x-seg model with custom data and I'm using BoTSORT for tracking on a 20-minute video.
I am experiencing ID switching issues. The model detects objects well, but when an object turns around or moves suddenly, the detection drops for a few seconds. When the object is redetected, it is assigned a new ID.

Interestingly, when I cut the video into 1-minute segments, the model performs perfectly without any ID issues. I've change the tracker paramaters and tracker.yaml paramaters multiple times but i haven't been able to achieve any improvement in performance.
 Has anyone encountered a similar problem with long-duration videos?

Here is my paramaters and tecs;

Model track code:

model_new = YOLO("1056imgsize/best.pt").to("cuda") // which is yolo26x-seg.pt

new_track = model_new.track(
    source=problem1_new,
    save=True,
    project="results",
    name="segmentation",
    stream=True,
    line_width=1,
    exist_ok=True,
    tracker=new_tracker,
    device="cuda",
    iou=0.25,
    conf=0.25,
    imgsz=1280,
)

for r in new_track:
    pass

tracker.yaml:

tracker_type: botsort

track_high_thresh: 0.1
track_low_thresh: 0.1 
new_track_thresh: 0.65
match_thresh: 0.5
track_buffer: 7500
fuse_score: True
gmc_method: none

with_reid: False
proximity_thresh: 0.5
appearance_thresh: 0.8
model: auto

**Comment by UltralyticsAssistant** [comment]
👋 Hello @barisatayy, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Thanks for sharing your YOLO26x-seg + BoTSORT setup and the `tracker.yaml` details—long-duration tracking + brief detection dropouts can surface edge cases like ID reassignment. An Ultralytics engineer will take a look soon 👀 (this is an automated response 🤖), and in the meantime the info below will help us reproduce what you’re seeing.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To help us narrow this down quickly, please add an MRE that includes:-A small video sample that reproduces the issue (for example: a short clip from the 20-minute video where ID switches happen, plus the “working” 1-minute segmented version if possible)-Your exact command/script (including how `new_tracker` is created/loaded)-The full console output and resulting saved artifacts (or key excerpts) from the run-Your environment details (OS, GPU, driver, CUDA, PyTorch, `ultralytics` version)

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
This looks like normal BoT-SORT behavior when detections drop: your 1‑minute clips “work” largely because the tracker state gets reset each run, while a 20‑minute run inevitably hits more drop/turn/occlusion events; also note your `track_low_thresh` == `track_high_thresh` disables the “second association” stage entirely, and `match_thresh=0.5` is quite strict, so lost tracks won’t re-link and you’ll get new IDs—try `conf=0.1` in `track()` (or omit it to use the tracking default), set `track_low_thresh < track_high_thresh` (e.g. `0.1/0.25`), relax `match_thresh` (e.g. `0.8`), and if you need IDs to survive turns/appearance changes enable `with_reid: True` in your BoT-SORT config as described in the [Tracking mode docs](https://docs.ultralytics.com/modes/track/).

**Reply by barisatayy** [reply]
Thank you for replying. 
I did what you said and i think that I need a custom reid model. There is a trained reid model but how can i use in my yolo track project? Should i change botsort.yaml file paramaters with "model : mytrainedmodel.pt" but it didnt work. How can i train my own reid model and use with yolo tracking?

**Reply by glenn-jocher** [reply]
`botsort.yaml` ReID only works with a YOLO *classification* model used as an embedding encoder (not a Detect/Segment model), so `model: mytrainedmodel.pt` will fail unless `mytrainedmodel.pt` is a `*-cls` checkpoint; this is implemented in `ultralytics.trackers.bot_sort.ReID` (see the [BoT-SORT tracker reference](https://docs.ultralytics.com/reference/trackers/bot_sort/)).

```yaml
# custom_botsort.yaml
tracker_type: botsort
with_reid: True
model: path/to/reid_best.pt   # must be a YOLO classification model (.pt/.engine), not your seg model
```

To train your own ReID encoder today, train a YOLO classification model on identity-labeled *crops* (one folder per identity, i.e. `train/<id_name>/*.jpg` and `val/<id_name>/*.jpg`), then point `model:` at the resulting `best.pt`:

```bash
yolo classify train model=yolo26n-cls.pt data=/path/to/reid_dataset imgsz=224 epochs=100
```

If it “didn’t work”, please paste the exact error traceback plus `yolo checks`, and confirm `tracker=` is a YAML file path (e.g. `tracker="custom_botsort.yaml"`) rather than a loaded dict/object.

**Reply by barisatayy** [reply]
Thanks for replying. 
I have worked what did you say but i couldnt get anywhere. Here is the issue that i have;
In my project there are many objects that look very similiar to each other. So i cant use reid model or train my own reid model. In the 1 or 2 minute video my yolo model with botsort works weel. But here is the issue that i couldnt solve, in the longer video id's changes. is this a yolo problem or should i use a different track algorithm? I tried bytesort, deepsort and deepocsort but botsort better then all. Even if im using botsort the problem couldnt solve. Also there no new objects are entring the camera and existing objects are not leaving. And my camera is fixed. What should i do?

**Reply by glenn-jocher** [reply]
With many near-identical objects, long-video ID churn after brief detection dropouts usually isn’t a YOLO issue but a fundamental limitation of online box+motion association (there’s simply not enough signal to “prove” which object is which once it’s missed); also note `track_buffer=7500` is frame-based and keeps lost tracks alive for minutes, which can actually *increase* wrong re-links over time—try setting it to just a few seconds worth of frames, and if you truly need perfect identity over 20 minutes the pragmatic solution is chunking with overlap and stitching IDs in post (the exact “lost/new track” logic is in the tracker update loop: see the [`BYTETracker.update` reference](https://docs.ultralytics.com/reference/trackers/byte_tracker/#method-ultralyticstrackersbyte_trackerbytetrackerupdate)); if you share a short problematic clip plus `yolo checks` output, I can sanity-check whether your current config is making re-links harder than necessary.
